cni/pkg/ambient/podutil.go

```diff
@@ -82,10 +82,9 @@ func AnnotateEnrolledPod(client kubernetes.Interface, pod *corev1.Pod) error {
 }
 
 func AnnotateUnenrollPod(client kubernetes.Interface, pod *corev1.Pod) error {
-	if pod.Annotations[constants.AmbientRedirection] != constants.AmbientRedirectionEnabled {
+	if pod.Annotations[constants.AmbientRedirection] != constants.AmbientRedirectionEnabled && pod.Annotations[constants.AmbientRedirection] != constants.AmbientRedirectionNone {
```

Comment:
I don't think this is right. The value can never be `"null"`, `annotationRemovePatch` removes the label, not sets it to the string `"null"`